pkg/harvester/mixins/harvester-vm/index.js

```diff
@@ -273,7 +273,7 @@ export default {
 
     needNewSecret() {
       // When creating a template it is always necessary to create a new secret.
-      return this.isCreate ? true : this.showYaml ? false : this.resourceType === HCI.VM_VERSION;
+      return this.showYaml ? false : this.resourceType === HCI.VM_VERSION || this.isCreate;
```

Comment:
Looks like a workaround, but should work in this case. 

nit: suggest reviewing the implementer of generateYaml, which should be immutable w/o changing any existing resource spec. Come out with an ultimate. 

[/pkg/harvester/edit/kubevirt.io.virtualmachine/index.vue#L533-L539](https://github.com/harvester/harvester-ui-extension/blob/467933bda0b60f2d8c12b4467661e198845f5480/pkg/harvester/edit/kubevirt.io.virtualmachine/index.vue#L533-L539)
```vue
    generateYaml() {
      this.showYaml = this?.$refs?.vmCruResource?.showYaml || false;
      this.parseVM();
      const out = saferDump(this.value);

      return out;
    },
```